### ISSUE TITLE
Update where OG images are pulled for sharing on Facebook

### ISSUE THREAD

**Issue by lizzycederberg** [Threespot/aldf-wp#234]
Right now if you share an image on Facebook it's pulling in the woodcut image, not the featured image. Please update to pull in the featured image on pages that contain a hero. It seems to be working on Twitter

ALDF would also like to know where these images are managed so they can manually update them if needed (especially for pages that do not contain a featured image). 

https://developers.facebook.com/tools/debug/sharing/?q=https%3A%2F%2Faldf.org%2Farticle%2Fanimal-legal-defense-fund-opposes-kristen-lindsey-appeal-to-texas-supreme-court%2F

![screen shot 2018-07-24 at 9 52 26 am](https://user-images.githubusercontent.com/13699106/43142837-5037a55a-8f27-11e8-9f2a-84f88f6803b7.png)

cc @cmontwill 

---
Checklist:
- 🔗  Provide links to affected pages
- 💁  Describe the issue and any steps required to reproduce it
- 📷  Include a screenshot or video of the problem
- 💻  List all versions of affected browsers and operating systems
- 🚩  Apply any relevant labels

**Comment by tedw** [comment]
Looks like Yoast will use the first image it finds by default and there is no way to customize that in the plugin GUI. I will have to manually tell it which images to use for each post type. This will probably take a couple of hours.

https://wordpress.stackexchange.com/questions/179447/is-there-a-filter-to-define-the-og-image-on-a-given-post

**Comment by tedw** [comment]
@lizzycederberg In the mean time, they can manually choose the social sharing image in the Yoast widget:

![yoast-social-image](https://user-images.githubusercontent.com/864716/43276541-f6608e54-90d2-11e8-9db0-80c679b871c5.png)
.

**Comment by tedw** [comment]
FWIW, it seems that Facebook is choosing that background image, not Yoast. The OG image on that [example page](https://aldf.org/project/florida-greyhound-racing-ban/) you posted is actually the correct [default image](https://aldf.org/wp-content/uploads/2018/07/ALDF_FB_Og_image.jpg).

Here's the generated markup:

```html
<meta property="og:image" content="https://aldf.org/wp-content/uploads/2018/07/ALDF_FB_Og_image.jpg" />
```

Perhaps that default image doesn’t comply with Facebook’s guidelines so FB is choosing the next image it finds.

**Comment by tedw** [comment]
FYI, ALDF will need to tell Facebook to re-scrape pages after updating:
https://developers.facebook.com/tools/debug/sharing

If they don’t do this, Facebook may take a few days to update.

**Comment by tedw** [comment]
Tried using this code but it didn’t seem to work:

```php
add_filter('wpseo_opengraph_image_size', function($size) {
    return 'large_size';
});
```

Docs https://yoast.com/wordpress/plugins/seo/api/

**Comment by lizzycederberg** [comment]
@tedw Update from David:

The issue is that the settings in Yoast aren’t actually being used for which image is displayed, even for pages where they are set in Yoast. You can see that here in the Facebook Object Debugger:
https://developers.facebook.com/tools/debug/og/object/?q=https%3A%2F%2Faldf.org%2Farticle%2Fshasta-county-suspends-contract-with-wildlife-killing-agency-after-pressure-from-the-animal-legal-defense-fund%2F
 
We used Yoast on the old site b/c we needed to be able to specify those images, typically adding a logo-badged image in the proper aspect ratio for both FB and Twitter. Currently, it’s not pulling those in, slowness aside…
 
We’d prefer to be able to use Yoast going forward. It didn’t slow down the old site. Can you look into if there is some incompatibility or conflict with the new site’s design that is preventing Yoast from working / slowing down the backend?

**Comment by tedw** [comment]
One issue is that Yoast uses the full-size image by default, which are often over Facebook’s 8 MB limit [docs](https://developers.facebook.com/docs/sharing/best-practices#images). I attempted to tell Yoast which image size to use the other day but the code had a typo: `'large_size'` should have been `'large'`.

I fixed this, and the code looks correct now, but for some reason Facebook is still not recognizing the image.

```html
<meta property="og:image" content="https://aldf.org/wp-content/uploads/2018/05/What-Tourists-Do-Not-See-1200x675.jpg" />
<meta property="og:image:secure_url" content="https://aldf.org/wp-content/uploads/2018/05/What-Tourists-Do-Not-See-1200x675.jpg" />
<meta property="og:image:width" content="1200" />
<meta property="og:image:height" content="675" />
<meta property="og:image:alt" content="sleddog" />
```

Facebook debugger:
https://developers.facebook.com/tools/debug/sharing/?q=https%3A%2F%2Faldf.org%2Farticle%2Fsled-dogs-reveals-the-cold-truth

I will keep looking into this but am out of ideas at the moment.
